apps/src/aiTutor/views/teacherDashboard/InteractionsTable.tsx

```diff
@@ -260,10 +261,13 @@ const InteractionsTable: React.FC<InteractionsTableProps> = ({sectionId}) => {
           labelText="Filter by Status"
           name="filter-statuses"
           onChange={handleStatusFilterChange}
-          onClearAll={() => setSelectedStatuses([])}
-          onSelectAll={() =>
-            setSelectedStatuses(Object.values(AITutorInteractionStatus))
-          }
+          selectAndClearAllConfig={{
+            onSelectAll: () =>
+              setSelectedStatuses(Object.values(AITutorInteractionStatus)),
+            onClearAll: () => setSelectedStatuses([]),
+            selectAllText: i18n.selectAll(),
+            clearAllText: i18n.clearAll(),
+          }}
```

Comment:
When you use unions, you add a type discriminator. If you coerce a type, it will change what types are available. For example, with the type `withControls`, it will enable the props `selectAll`, `clearAll`, `onSelectAll`, `onClearAll`, etc. However, if you use the type `withoutControls`, then if you attempt to use those props, it will error out because those props aren't needed.

Take a look at this [commit](https://github.com/code-dot-org/code-dot-org/commit/87746c735f461af54261c3f1ff3646807486d436) (Which isn't fully polished, I'm not sure if it will pass drone) but it should help illustrate how we can use unions.